dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/DataQueryParams.java

```diff
@@ -1446,6 +1440,45 @@ public boolean hasStartEndDate()
         return startDate != null && endDate != null;
     }
 
+    /**
+     * Indicates whether this query has a nonempty dateRangeList.
+     */
+    public boolean hasDateRangeList()
+    {
+        return dateRangeList != null && dateRangeList.size() > 0;
```

Comment:
I think would be safe to use "!dataRangeList.isEmpty()".
Otherwise, line 677 could raise NPE as well.

Personally, I usually prefer CollectionUtils.isNotEmpty(). It's up to you.

Reply:
done